examples/datapipe.py

```diff
@@ -25,18 +32,36 @@ def molecule_datapipe() -> IterDataPipe:
     root_dir = osp.join(osp.dirname(osp.realpath(__file__)), '..', 'data')
     path = download_url(f'{url}/HIV.csv', root_dir)
 
-    datapipe = FileOpener([path])
-    datapipe = datapipe.parse_csv_as_dict()
+    datapipe = FileOpener([path], mode="rt")
+    # Convert CSV rows into dictionaries, skipping the header row
+    datapipe = datapipe.map(lambda file: (
+        dict(zip(["smiles", "activity", "HIV_active"], row))
+        for i, row in enumerate(csv.reader(file[1])) if i > 0 and row))
+
+    datapipe = IterableWrapper(chain.from_iterable(datapipe))
     datapipe = datapipe.parse_smiles(target_key='HIV_active')
-    datapipe = datapipe.in_memory_cache()  # Cache graph instances in-memory.
 
+    _, cached_datapipe = tee(datapipe)
+    datapipe = IterableWrapper(cached_datapipe)
     return datapipe
 
 
 @torch.utils.data.functional_datapipe('read_mesh')
 class MeshOpener(IterDataPipe):
     # A custom DataPipe to load and parse mesh data into PyG data objects.
     def __init__(self, dp: IterDataPipe):
+        import importlib.util
+        installed = True
+        for package in ['meshio', 'torch_cluster']:
+            if importlib.util.find_spec(package) is None:
+                installed = False
+                print(
+                    f"This example requires the package {package} to be installed."
+                )
+                print(f"Please run: 'pip install {package}'")
+        if not installed:
+            exit()
```

Comment:
Let's raise an exception so that it exits with non-zero status

Reply:
Can we put this at the top of this script? As a user, I'd like the script fail early if necessary packages are unavailable.